emu/src/cpu/arm7tdmi.rs

```diff
@@ -1094,6 +1098,35 @@ impl Arm7tdmi {
         }
     }
 
+    pub(crate) fn multiple_load_store(
+        &mut self,
+        load_store: LoadStoreKind,
+        base_register: u16,
+        register_list: u16,
+    ) -> Option<u32> {
+        match load_store {
+            LoadStoreKind::Store => unimplemented!("multiple store"),
+            LoadStoreKind::Load => {
+                let base_address = self.registers.register_at(base_register as usize);
+                let mut register_one = 0;
+                for r in 0..=7 {
+                    if register_list.get_bit(r) {
+                        let address = base_address + (register_one as u32 * 4);
+                        let value = self.memory.lock().unwrap().read_word(address as usize);
+                        self.registers.set_register_at(r as usize, value);
+                        register_one += 1;
+                    }
+                }
+
+                let new_base_address = base_address + (register_list.count_ones() * 4);
+                self.registers
+                    .set_register_at(base_register as usize, new_base_address);
```

Comment:
I know I'm nitpicking but I think it's more clear if we do something like:

```rust
let mut address = self.registers.register_at(...);

for ... {
  if ... {
     let value = ...;
     self.registers.set_registe_at(...);
     address += 4;
   }
  }

self.registers.set_register_at(...);
```

Or better, we could reuse the logic in the function `exec_data_transfer` in `arm7tdmi.rs`

Reply:
Now is more easy to read :)